File: samples/go/08objects/circle/circle.go
package circle

import (
	"fmt"
	"math"
)

type Circle struct {
	radius float64
}

func New(radius float64) Circle {
	return Circle{
		radius: radius,
	}
}

func Area(c Circle) float64 {
	return math.Pi * c.radius * c.radius
}

func Perimeter(c Circle) float64 {
	return 2 * math.Pi * c.radius
}

func GetRadius(c Circle) float64 {
	return c.radius
}

func Print(c Circle) {
	fmt.Printf("[Circle] Area: %f Perimeter: %f\n",
		Area(c),
		Perimeter(c))
}

File: samples/java/08objects/main/Circle.java
package main;

import java.lang.Math;
import main.Shape;

public class Circle extends Shape {
    private double radius;

    public Circle(float radius) {
        this.radius = radius;
    }

    public double area() {
        System.out.println(this.getClass().getName());
        return Math.PI * this.radius * this.radius;
    }

    public double perimeter() {
        return 2.0 * Math.PI * this.radius;
    }
}
